subworkflows/local/annotate_mt_snvs.nf

```diff
@@ -0,0 +1,97 @@
+//
+// Annotate MT
+//
+
+include { TABIX_TABIX as TABIX_TABIX_MT                         } from '../../modules/nf-core/tabix/tabix/main'
+include { ENSEMBLVEP as ENSEMBLVEP_MT                           } from '../../modules/local/ensemblvep/main'
+include { HAPLOGREP2_CLASSIFY as HAPLOGREP2_CLASSIFY_MT         } from '../../modules/nf-core/haplogrep2/classify/main'
+include { VCFANNO as VCFANNO_MT                                 } from '../../modules/nf-core/vcfanno/main'
+include { ANNOTATE_CADD                                         } from './annotation/annotate_cadd'
+include { TABIX_BGZIPTABIX as ZIP_TABIX_HMTNOTE                 } from '../../modules/nf-core/tabix/bgziptabix/main'
+include { HMTNOTE_ANNOTATE                                      } from '../../modules/nf-core/hmtnote/annotate/main'
+
+workflow ANNOTATE_MT_SNVS {
+    take:
+        ch_mt_vcf              // channel: [mandatory] [ val(meta), path(vcf) ]
+        ch_mt_tbi              // channel: [mandatory] [ val(meta), path(tbi) ]
+        ch_cadd_header         // channel: [mandatory] [ path(txt) ]
+        ch_cadd_resources      // channel: [mandatory] [ path(annotation) ]
+        ch_genome_fasta        // channel: [mandatory] [ val(meta), path(fasta) ]
+        ch_vcfanno_resources   // channel: [mandatory] [ path(resources) ]
+        ch_vcfanno_toml        // channel: [mandatory] [ path(toml) ]
+        val_vep_genome         // string:  [mandatory] GRCh37 or GRCh38
+        val_vep_cache_version  // string:  [mandatory] 107
+        ch_vep_cache           // channel: [mandatory] [ path(cache) ]
+
+    main:
+        ch_versions = Channel.empty()
+
+        // Annotating with CADD
+        ANNOTATE_CADD (
+            ch_mt_vcf,
+            ch_mt_tbi,
+            ch_cadd_header,
+            ch_cadd_resources
+        )
+
+        // Pick input for vep
+        ch_mt_vcf
+            .combine(ANNOTATE_CADD.out.vcf.ifEmpty("null"))
+            .branch { it  ->
+                merged: it[2].equals("null")
+                    return [it[0], it[1]]
+                cadd: !(it[2].equals("null"))
+                    return [it[2], it[3]]
+            }
+            .set { ch_for_mix }
+        ch_vep_in = ch_for_mix.merged.mix(ch_for_mix.cadd)
```

Comment:
Is there reason to use the same type of channel combinations as in https://github.com/nf-core/raredisease/pull/421 ?